daq/faucet_state_collector.py

```diff
@@ -165,9 +165,42 @@ def get_switch_raw(self, switch_name):
     def get_stack_topo(self):
         """Returns formatted topology object"""
         topo_map = {}
-        topo_obj = self.topo_state
         with self.lock:
-            for link in topo_obj.get(TOPOLOGY_GRAPH, {}).get("links", []):
+            config_obj = self.system_states.get(FAUCET_CONFIG, {}).get(DPS_CFG, {})
+            links = self.topo_state.get(TOPOLOGY_GRAPH, {}).get("links", [])
+            path_to_root = self.topo_state.get(ROOT_PATH, {})
+            for dp, dp_obj in config_obj.items():
+                for iface, iface_obj in dp_obj.get("interfaces", {}).items():
+                    dp_s = iface_obj.get("stack", {}).get("dp")
+                    port_s = str(iface_obj.get("stack", {}).get("port"))
+                    if dp_s and port_s:
+                        link_obj = {}
+                        if dp+":"+iface < dp_s+":"+port_s:
```

Comment:
this could use a clean-up, there's too much duplicated logic/stuff here.  E.g., create intermediate keys based on the ordering, and then you can just do link_obj[start_key] = dp, where start_key is "switch_a" or "switch_b",  Also the dp+":"+iface should only be done once and stored in an intermediate variable. Play around with it, and if you really get stuck let me know!